src/Avalonia.Controls/ComboBox.cs

```diff
@@ -0,0 +1,192 @@
+                nameof(Text),
+                o => o._text,
+                (o, v) => o.Text = v);
+
+        private string _text;
+        private bool _isDropDownOpen;
+        private Popup _popup;
+        private TextBox _textBox;
+
+        /// <summary>
+        /// Initializes static members of the <see cref="ComboBox"/> class.
+        /// </summary>
+        static ComboBox()
+        {
+            FocusableProperty.OverrideDefaultValue<ComboBox>(true);
+            SelectedItemProperty.Changed.AddClassHandler<ComboBox>(x => x.SelectedItemChanged);
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the dropdown is currently open.
+        /// </summary>
+        public bool IsDropDownOpen
+        {
+            get { return _isDropDownOpen; }
+            set { SetAndRaise(IsDropDownOpenProperty, ref _isDropDownOpen, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum height for the dropdown list.
+        /// </summary>
+        public double MaxDropDownHeight
+        {
+            get { return GetValue(MaxDropDownHeightProperty); }
+            set { SetValue(MaxDropDownHeightProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the text displayed in the <see cref="ComboBox"/>.
+        /// </summary>
+        public string Text
+        {
+            get { return GetValue(TextProperty); }
+            set { SetAndRaise(TextProperty, ref _text, value); }
+        }
+
+        /// <inheritdoc/>
+        protected override IItemContainerGenerator CreateItemContainerGenerator()
+        {
+            return new ItemContainerGenerator<ComboBoxItem>(
+                this,
+                ComboBoxItem.ContentProperty,
+                ComboBoxItem.ContentTemplateProperty);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnAttachedToLogicalTree(LogicalTreeAttachmentEventArgs e)
+        {
+            base.OnAttachedToLogicalTree(e);
+            this.UpdateSelectionBoxText(this.SelectedItem);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnKeyDown(KeyEventArgs e)
+        {
+            base.OnKeyDown(e);
+
+            if (!e.Handled)
+            {
+                if (e.Key == Key.F4 ||
+                    (e.Key == Key.Down && ((e.Modifiers & InputModifiers.Alt) != 0)))
+                {
+                    IsDropDownOpen = !IsDropDownOpen;
+                    e.Handled = true;
+                }
+                else if (IsDropDownOpen && (e.Key == Key.Escape || e.Key == Key.Enter))
+                {
+                    IsDropDownOpen = false;
+                    e.Handled = true;
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void OnPointerPressed(PointerPressedEventArgs e)
+        {
+            if (!IsDropDownOpen && ((IVisual)e.Source).GetVisualRoot() != typeof(PopupRoot))
```

Comment:
The drop down should be opened by the toggle button, not when clicking anywhere else, so I think this check is redundant.